### PR TITLE
doc: Add PHP section to index

### PR DESCRIPTION
**Pull request by etu**
###### Motivation for this change
I guess this should work? We have the markdown file from before and I guess that is converted to xml somehow?

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by Ma27 — CHANGES_REQUESTED** [review]
The approach is right since the `.md`-files are converted to docbook using `pandoc` first.

However, `nix-build doc/` is breaking with this change due to a XML-id collision: `python.section.md` and `php.section.md` have a `User guide`-headline which is transformed into an `xml:id` named `user-guide` (twice).

It may be possible to work around this in `pandoc` (in `doc/Makefile`) using `--id-prefix`, the easier fix would be however to replace the `User guide`-headline in `php.section.md` by something else I guess.

**Comment by etu** [comment]
TIL how to build the documentation locally!

I have made some changes to our headlines and tested it locally :-)

**Comment by flokli** [comment]
I tried updating [`doc/Makefile` to use `--id-prefix`](https://github.com/flokli/nixpkgs/commit/f9c10d4a69b22d842c7ad3db0c40abd7f88960f3), but unfortunately, this also prefixes all links to things outside each specific `.md` file.

I'll ask upstream pandoc if there's a way around this, but I assume for now we need to chase duplicates manually :-/

**Comment by etu** [comment]
> I'll ask upstream pandoc if there's a way around this, but I assume for now we can't namespace `.md` documentation, and need to be super careful with how we name our headings :-/

That means that we shouldn't make headings like `Overview` because someone else would probably want to have that as well. But prefixing with something like `$lang overview` should be fairly safe since it's kinda namespaced to one of the files.

File: doc/languages-frameworks/php.section.md [hunk 1]
```diff
@@ -119,3 +119,20 @@ with the extensions `imagick` and `opcache` enabled:
...
+enough and you may want to extend it.
+
+A common case of that is the `composer` package since a project may
+depend on certain extensions and then `composer` won't work very well
+for that project unless it has those extensions loaded.
+
+Example of building `composer` with extra extensions added:
+```nix
+(php.withExtensions ({all, enabled }:
+  enabled ++ (with all; [ imagick redis ]))
```
Comment (talyz) [inline doc/languages-frameworks/php.section.md:136]:
```suggestion
All interactive tools use the PHP package you get them from, so all
packages at `php.packages.*` use the `php` package with its default
extensions. Sometimes this default set of extensions isn't enough and
you may want to extend it. A common case of this is the `composer`
package: a project may depend on certain extensions and `composer`
won't work with that project unless those extensions are loaded.

Example of building `composer` with additional extensions:
```nix
(php.withExtensions ({ all, enabled }:
  enabled ++ (with all; [ imagick redis ]))
```

Comment (etu) [reply doc/languages-frameworks/php.section.md]: Added :)

File: doc/languages-frameworks/php.section.md [hunk 2]
```diff
@@ -1,10 +1,8 @@
 # PHP
```
Comment (jtojnar) [inline doc/languages-frameworks/php.section.md:1]: There is a pandoc extension, not sure if enabled by default, that allows defining ids. All headings need to have IDs specified to maintain permanent reference.

```suggestion
# PHP {#sec-language-php}
```

File: doc/languages-frameworks/php.section.md [hunk 3]
```diff
@@ -1,10 +1,8 @@
...
 
-## User Guide
+## Using PHP
```
Comment (jtojnar) [inline doc/languages-frameworks/php.section.md:3]:
```suggestion
## User Guide {#ssec-php-user-guide}
```

**Review by jtojnar — CHANGES_REQUESTED** [review]

File: doc/languages-frameworks/php.section.md (hunk 2 above)
Comment (etu) [reply doc/languages-frameworks/php.section.md]: Seems to work, added to all headings in the document :)

File: doc/languages-frameworks/php.section.md [hunk 4]
```diff
@@ -36,7 +34,7 @@ opcache extension shipped with PHP is available at
 `php.extensions.opcache` and the third-party ImageMagick extension at
 `php.extensions.imagick`.
 
-#### Installing PHP with extensions
+### Installing PHP with extensions {#sssec-php-user-guide-installing-with-extensions}
```
Comment (jtojnar) [inline doc/languages-frameworks/php.section.md:37]:
```suggestion
### Installing PHP with extensions {#ssec-php-user-guide-installing-with-extensions}
```

File: doc/languages-frameworks/php.section.md [hunk 5]
```diff
@@ -89,14 +87,14 @@ php.buildEnv {
 }
 ```
 
-##### Example setup for `phpfpm`
+#### Example setup for `phpfpm` {#ssssec-php-user-guide-installing-with-extensions-phpfpm}
```
Comment (jtojnar) [inline doc/languages-frameworks/php.section.md:90]:
```suggestion
#### Example setup for `phpfpm` {#sssec-php-user-guide-installing-with-extensions-phpfpm}
```

Comment (jtojnar) [reply doc/languages-frameworks/php.section.md:90]: Or maybe just

```suggestion
#### Example setup for `phpfpm` {#ssec-php-user-guide-installing-with-extensions-phpfpm}
```

sub-sub-section is still a sub-section after a fashion.

**Review by jtojnar — APPROVED** [review]

**Review by talyz — APPROVED** [review]

**Comment by talyz** [comment]
After looking at the rendered documentation a bit closer, I think it would be good to just get rid of the `User Guide` heading, since there's no other header at the same level anyway. What do you think?

**Comment by etu** [comment]
My thought is that it can be good to keep, the reasoning behind that is that I in the future would like to see that we manage to get something in to build things with composer.

And that would be it's own headline among the lines `Developing guide` on the same level as `User guide`.

But I already dropped one level in this PR :smile:

**Comment by talyz** [comment]
Okay, yeah, that makes sense. I guess we could also document `buildPecl` and such, which should probably be under a different heading on that level. :)

**Pull request merged**